src/apps/chatter/hooks/use-room.js

```diff
@@ -34,38 +39,47 @@ export function useRoom (roomId) {
       console.log(e.entry)
     }
     socketRef.current.onclose = () => {
-      console.log('WebSocket closed let\'s reopen')
-      connect()
+      console.log('WebSocket closed')
+      // connect()
     }
   }
 
-  const onNewMessage = (data) => {
+  const onNewMessage = (messageString) => {
     // console.log('DATA IN: ', { data })
-    const parsedData = JSON.parse(data)
-    const command = parsedData.command
+    const parsedMessage = JSON.parse(messageString)
+    const command = parsedMessage.command
     if (command === NEW_ENTRY) {
-      onNewEntry(parsedData)
+      onNewEntry(parsedMessage)
     } else if (command === ENTRIES) {
-      onFetchEntries(parsedData)
+      onFetchEntries(parsedMessage)
+    } else if (command === INIT_RESPONSE) {
+      onInitRespose(parsedMessage)
     }
   }
 
   // Functions called upon input from the server
-  const onNewEntry = (parsedData) => {
-    setEntries((previous) => ([...previous, parsedData.entry]))
+  const onNewEntry = (parsedMessage) => {
+    setEntries((previous) => ([...previous, parsedMessage.entry]))
   }
 
-  const onFetchEntries = (parsedData) => {
-    // console.log('RECEIVED NEW MESSAGES COMMAND: ', parsedData)
-    setEntries(parsedData.entries.reverse())
+  const onFetchEntries = (parsedMessage) => {
+    // console.log('RECEIVED NEW MESSAGES COMMAND: ', parsedMessage)
+    setEntries(parsedMessage.entries.reverse())
+  }
+
+  const onInitRespose = (parsedMessage) => {
+    console.log('RECEIVED INIT RESPONSE COMMAND: ', parsedMessage)
+    setIsAuthorized(parsedMessage.authorized)
+    setIsInitialized(true)
```

Comment:
Could this be combined into a single state reference?

If you init a room that does not require is a pass, then `isAuthrorized = true`
If you init a room that **does** require a pass, then it will be true or false based on given password.

Doesn't appear to be a case where you would call `setIsInitialized` without also calling  `setIsAuthorized`?

Reply:
The way it works currently is that the way the front end figures out if a room is password protected is to initially send a password of "" to every room. If this works, then INIT_RESPONSE command will come back with authorized: true. At that point, is initialized and isAuthorized are both set to true at the same time.
Conversely, for password protected rooms, initialization will be set to true but isAuthorized will remain false, which causes the password field to show.
There may be a smarter way to do this that probably involves pulling down a boolean for each room that identifies weather a password is needed for that room, but I would have to puzzle over it for a bit.
The reason blank passwords are used is because I recall reading somewhere that empty strings should be used instead of null in model fields? Not sure why.

Reply:
That's a nope to empty strings instead of null.

Also, you could fetch the messages, if the room is protected it will fail, and when the front-end gets the error message, it can interpret it, if it's because of a password, then show the form?